common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -885,3 +885,41 @@ def get_username(strategy, details, backend, user=None, *args, **kwargs):
     else:
         final_username = storage.user.get_username(user)
     return {'username': final_username}
+
+
+def set_learner_active_enterprise(user=None, backend=None, response=None, *args, **kwargs):
+    """
+    Check user active enterprise, if it matches to idp-enterprise then no action needed otherwise make the
+    idp enterprise active via post_learner_active_enterprise method.
+    """
+    from openedx.features.enterprise_support.api import (
+        EnterpriseApiClient,
+        EnterpriseApiException,
+        get_enterprise_learner_data,
+    )
+    if backend.name == 'tpa-saml':
```

Comment:
Can we make a util of this method and use it in all the places in the file ... https://github.com/edx/edx-platform/blob/master/common/djangoapps/third_party_auth/pipeline.py#L556